spacy/language.py

```diff
@@ -2328,13 +2328,18 @@ def restore(self) -> None:
         self[:] = []
 
 
-def _copy_examples(examples: Iterable[Example]) -> List[Example]:
+def _copy_examples(
+    examples: Iterable[Example], copy_x: bool = True, copy_y: bool = False
+) -> List[Example]:
     """Make a copy of a batch of examples, copying the predicted Doc as well.
     This is used in contexts where we need to take ownership of the examples
     so that they can be mutated, for instance during Language.evaluate and
     Language.update.
     """
-    return [Example(eg.x.copy(), eg.y) for eg in examples]
+    return [
+        Example(eg.x.copy() if copy_x else eg.x, eg.y.copy() if copy_y else eg.y)
+        for eg in examples
+    ]
```

Comment:
I wonder if we should emit a warning here if we do a no-op, i.e., both `copy_x` and `copy_y` are `False`. I can't think of a reason why the user would call the function at all in such a case, but it feels weird to have a function that basically goes `def do_something(really_do_something: bool = True):`?

Reply:
It's an internal function. Besides that, I don't think it's a no-op since it clears things like cached alignments.